modules/nextflow/src/main/groovy/nextflow/scm/RepositoryProvider.groovy

```diff
@@ -65,7 +65,7 @@ abstract class RepositoryProvider {
     /**
      * The name of the commit/branch/tag
      */
-    protected String revision
+    String revision
```

Comment:
Instead of changing to public better to a getter to maintain the encapsulation. May even it be changed to `private visibility` (maybe). 

I would also add `project` and `config` for consistency

Reply:
Maybe my PR title is a little confusing, what I mean is enable public access the 'groovy way' via a property, which makes the underlying field private, not explicitly make the field public. i.e. taking advantage of groovy's accessor auto-generation/usage.

Reply:
understand, my point is to keep protected (or private) and only add public getter so it can be accessed only for read but not for write

Reply:
`revision` already has a public setter `RepositoryProvider setRevision(String revision) `.  `project` and `config` do not, rather they are populated via the constructor, naturally, as they do not change as `revision` could. The current change to omit the modifer results in:

- private underlying field
- autogenerated public getter
- pre-existing explicitly defined public setter 

I will add public getters for project and config also.

Reply:
Got your point still better keeping it `protected` and add `getRevision` for consistency with other getters

Reply:
Ah, <click> gotcha :)

Reply:
done

Reply:
ty!